### DISCUSSION TITLE
What OpenAPIs already exist?

### DISCUSSION THREAD

**Discussion by kinlane** [OAI/sig-travel#7] [General]
What existing OpenAPIs and Swagger files already exist for existing API providers, organizations, and standards across the travel industry?

**Comment by swaldron58** [comment]
I will need a clarification of what constitutes or defines an "OpenAPI". Many claim to have published an Open API with little to no justification of the use of the term. Does just using JSON make it open?

**Reply by kinlane** [reply]
I am looking purely for formal OpenAPI 3.x spec or Swagger 2.0. For this group the minimum bar for discussion will be an OpenAPI spec, so it doesn't refer to the "openness" of the API, but specifically a machine readable contract.

**Reply by swaldron58** [reply]
OK, I'll get Brain's team to help (LinksRez). On the portal for Open Travel they have been working on there is a page showing what APIs are available in the portal which all should meet your stated criteria. The rest will be harder to sort out. For example all the TravelPort newer APIs called Trip Services would qualify, but as I don't work there anymore I can't provide a list. 
In the past I have run across services that queried web servers to get a response on what software and version they were running. I have looked for the same on APIs. I wanted to get an idea of how many travel related APIs were out there. Even better if I could get some idea of the type. But I have not found any source anywhere near complete.   Suggestions? If we could get a list of URLs, at least the domain name, we may be able to discover more.

**Comment by swaldron58** [comment]
I am cleaning up documents where Open Travel workgroups have documented message exchange patterns for hospitability. These show the URLs for Get/Post/Put calls that would be accepted as a standard approach.

**Reply by kinlane** [reply]
I have an [OpenTravel OpenAPI published here](https://www.postman.com/api-evangelist/workspace/opentravel/api/29bad90d-907e-4b6b-9255-717cca94a0fd?version=418f07c8-0dda-4a79-b5f3-2bfff81a7de9&tab=define). I have no idea as to its completeness. I understand that the OpenAPI is generated from the model based upon which ones are selected, so it might be nice to break down into separate files by domain.

**Comment by kinlane** [comment]
I feel like OpenTravel is going to be where much of our conversations begin as part of this working group.

**Reply by swaldron58** [reply]
Open Travel and thru that into the MDD approach. It would be important for all involved to really understand what a domain model approach entails. It's not a shared data dictionary though sometimes called that. It's a shared model directly embedded in the development process.

**Comment by luisweir** [comment]
@kinlane you already have this but just for the record our Open APIs and Collections are here

https://github.com/oracle/hospitality-api-docs
